Stop queuing due to decompositions from within `qml.matrix`. (#7976)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Bug fixes 🐛</h3>

> * Fixes a bug that caused operators to be queued from within :func:`~.matrix` if it required
>   decomposition of an operator that in turn queues its constituents in its decomposition method.

Reviewer: ```suggestion
* Fixes a bug in :func:`~.matrix` where an operator's
  constituents were incorrectly queued if its decomposition was requested.
```
